api/envoy/config/filter/http/ext_authz/v2alpha/ext_authz.proto

```diff
@@ -68,23 +77,52 @@ message HttpService {
   // Sets the HTTP server URI which the authorization requests must be sent to.
   envoy.api.v2.core.HttpUri server_uri = 1;
 
-  // Sets an optional prefix to the value of authorization request header *Path*.
-  string path_prefix = 2;
-
+  reserved 2;
   reserved 3;
+  reserved 4;
+  reserved 5;
+  reserved 6;
+
+  // Settings for controlling request headers forwarded from the filter to the authorization server.
+  AuthorizationRequest authorization_request = 7;
+
+  // Settings for controlling authorization response forwarded from the filter to a client,
+  // or to an upstream service.
+  AuthorizationResponse authorization_response = 8;
+}
+
+message AuthorizationRequest {
+  // Sets an optional prefix to the value of authorization request header *Path*.
+  string path_prefix = 1;
 
-  // Sets a list of headers that can be sent from the authorization server to the upstream service,
-  // or to the downstream client when present in the authorization response. Note that a matched
-  // request header will have its value overridden by the ones sent from the authorization server.
-  repeated string allowed_authorization_headers = 4;
+  // Sets a list of keys that are used to determine which client request headers should
+  // be forwarded *from the filter* to the authorization server. Note that *Content-Length*,
+  // *Authority*, *Method*, *Path* and *Authorization* are always dispatched to the authorization
+  // server by default. The message will not contain body data and the *Content-Length* will be set
+  // to zero.
+  repeated string allowed_headers = 2;
```

Comment:
Can we just do a repeated list of StringMatcher here? Then we don't need an exact match and a prefix match list? Same elsewhere? StringMatcher for all matches for flexibility?

Reply:
Yes, I like the idea of having less lists if we can. I've done some work already and it looks cleaner.

Reply:
Ping on this?